structuredevent-service-cdp/src/main/java/com/sequenceiq/cloudbreak/structuredevent/service/db/CDPStructuredEventDBService.java

```diff
@@ -68,43 +69,43 @@ protected void prepareCreation(CDPStructuredEventEntity resource) {
             LOGGER.warn(msg, ex);
             throw new CloudbreakServiceException(msg, ex);
         }
     }
 
     @Override
-    public <T extends CDPStructuredEvent> Page<T> getPagedEventsOfResource(List<StructuredEventType> eventType, String resourceCrn, Pageable pageable) {
-        LOGGER.debug("Gathering pageable events for type: '{}' and resource CRN: '{}'", eventType, resourceCrn);
+    public <T extends CDPStructuredEvent> List<T> getEventsOfResource(List<StructuredEventType> eventTypes, String resourceCrn) {
+        LOGGER.debug("Gathering events for type: '{}' and resource CRN: '{}'", eventTypes, resourceCrn);
+        List<StructuredEventType> types = getAllEventTypeIfEmpty(eventTypes);
         try {
-            Page<CDPStructuredEventEntity> events = pagingStructuredEventRepository.findByEventTypeInAndResourceCrn(eventType, resourceCrn, pageable);
-            return (Page<T>) Optional.ofNullable(events).orElse(Page.empty()).map(event -> conversionService.convert(event, CDPStructuredEvent.class));
-        } catch (Exception ex) {
-            String msg = String.format("Failed get pageable events for type: '%s' and resource CRN: '%s'", eventType, resourceCrn);
-            LOGGER.warn(msg, ex);
-            throw new CloudbreakServiceException(msg, ex);
-        }
-    }
-
-    @Override
-    public <T extends CDPStructuredEvent> List<T> getNotificationEventsOfResource(StructuredEventType eventType, String resourceCrn) {
-        LOGGER.debug("Gathering events for type: '{}' and resource CRN: '{}'", eventType, resourceCrn);
-        try {
-            List<CDPStructuredEventEntity> events = structuredEventRepository.findByEventTypeAndResourceCrn(eventType, resourceCrn);
+            List<CDPStructuredEventEntity> events = structuredEventRepository.findByEventTypeInAndResourceCrn(types, resourceCrn);
             return (List<T>) Optional.ofNullable(events).orElse(new ArrayList<>()).stream()
                     .map(event -> conversionService.convert(event, CDPStructuredEvent.class))
                     .collect(Collectors.toList());
         } catch (Exception ex) {
-            String msg = String.format("Failed get events for type: '%s' and resource CRN: '%s'", eventType, resourceCrn);
+            String msg = String.format("Failed get events for types: '%s' and resource CRN: '%s'", types, resourceCrn);
             LOGGER.warn(msg, ex);
             throw new CloudbreakServiceException(msg, ex);
         }
     }
+
+    private List<StructuredEventType> getAllEventTypeIfEmpty(List<StructuredEventType> eventTypes) {
+        List<StructuredEventType> types = new ArrayList<>(eventTypes);
+        if (CollectionUtils.isEmpty(eventTypes)) {
+            LOGGER.info("We need to add all structured event types to the filter");
+            types.add(StructuredEventType.NOTIFICATION);
+            types.add(StructuredEventType.REST);
+            types.add(StructuredEventType.FLOW);
+        }
```

Comment:
Shouldn't we do something if the eventypes is not empty??

Reply:
Yes, we do. We return with the original list